### ISSUE TITLE
Performance Issue

### ISSUE THREAD

**Issue by Justin9606** [hoaphantn7604/react-native-curved-bottom-bar#23]
I have started using this package but Currently, I am having a performance issue when I switch the tab it is pretty long time to switch the tab. It is very recognizable slow.

Has anyone had similar issue? or any solution?

Thanks

**Comment by hoaphantn7604** [comment]
hi @Justin9606 ,
I use useMemo to increase performance.

The application has applied this package.
CH Play: https://play.google.com/store/apps/details?id=com.vngame.kingenglish
App store: https://apps.apple.com/us/app/king-english-game/id1601137730

```js
 const renderMain = useMemo(() => {
    return (
      <CurvedBottomBar.Navigator
        style={styles.bottomBar}
        strokeWidth={0.5}
        height={scale(isIOS && hasNotch ? 60 : 50)}
        circleWidth={scale(isIOS ? 60 : 50)}
        bgColor={COLORS.PRIMARY}
        initialRouteName="title0"
        borderTopLeftRight
        swipeEnabled={true}
        renderCircle={({ selectedTab, navigate }) => (
          <Animated.View style={styles.btnCircle}>
          </Animated.View>
        )}
        tabBar={renderTabBar}>
        <CurvedBottomBar.Screen
          name="title1"
          position="left"
          component={({ navigate }) => <HomeScreen />}
        />
        <CurvedBottomBar.Screen
          name="title2"
          component={({ navigate }) => <SettingScreen />}
          position="right"
        />
      </CurvedBottomBar.Navigator>
    );
  }, []);
```

**Comment by Justin9606** [comment]
> hi @Justin9606 , I use useMemo to increase performance.
> 
> The application has applied this package. CH Play: https://play.google.com/store/apps/details?id=com.vngame.kingenglish App store: https://apps.apple.com/us/app/king-english-game/id1601137730
> 
> ```js
>  const renderMain = useMemo(() => {
>     return (
>       <CurvedBottomBar.Navigator
>         style={styles.bottomBar}
>         strokeWidth={0.5}
>         height={scale(isIOS && hasNotch ? 60 : 50)}
>         circleWidth={scale(isIOS ? 60 : 50)}
>         bgColor={COLORS.PRIMARY}
>         initialRouteName="title0"
>         borderTopLeftRight
>         swipeEnabled={true}
>         renderCircle={({ selectedTab, navigate }) => (
>           <Animated.View style={styles.btnCircle}>
>           </Animated.View>
>         )}
>         tabBar={renderTabBar}>
>         <CurvedBottomBar.Screen
>           name="title1"
>           position="left"
>           component={({ navigate }) => <HomeScreen />}
>         />
>         <CurvedBottomBar.Screen
>           name="title2"
>           component={({ navigate }) => <SettingScreen />}
>           position="right"
>         />
>       </CurvedBottomBar.Navigator>
>     );
>   }, []);
> ```

Thank you. I have checked your app's tab switch is really faster than ours We will use also useMemo but if you have any other suggestion pls do let us know.
Thank you

**Comment by hoaphantn7604** [comment]
Hi,
This issue has been resolved in version 2.0.0.
